error.go

```diff
@@ -701,5 +702,9 @@ func (err WriteErrors) Count() int {
 }
 
 func (err WriteErrors) Error() string {
-	return fmt.Sprintf("kafka write errors (%d/%d)", err.Count(), len(err))
+	var sb strings.Builder
+	for i, writeError := range err {
```

Comment:
Can write error ever be nil here?

Reply:
i don't think so, if err is empty then it will never execute it what do you think? @rhansen2

Reply:
It is not possible for one element of the array to be nil?

Reply:
Based on what's documented [here](https://pkg.go.dev/github.com/segmentio/kafka-go#WriteErrors) I believe we have to handle the case when an item in the array is nil.

Reply:
@rhansen2 thanks you are right, I added validation of that.